agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -631,6 +631,114 @@ func fluentdDriverTest(taskDefinition string, t *testing.T) {
 	assert.NoError(t, err, "failed to find the log tag specified in the task definition")
 }
 
+// TestLogDriverSecretSupport tests the log driver secret support using
+// the awslogs and awslogs-multiline-pattern option is utilized as a secret storing as an encrypted parameter
+func TestLogDriverSecretSupport(t *testing.T) {
+	RequireDockerVersion(t, ">=17.06.2")
+
+	if os.Getenv("TEST_DISABLE_EXECUTION_ROLE") == "true" {
+		t.Skip("TEST_DISABLE_EXECUTION_ROLE was set to true")
+	}
+
+	if IsCNPartition() {
+		t.Skip("Skip TestLogDriverSecretSupport in China partition")
+	}
+
+	cwlClient := cloudwatchlogs.New(session.New(), aws.NewConfig().WithRegion(*ECS.Config.Region))
+	// Test whether the log group exists or not
+	respDescribeLogGroups, err := cwlClient.DescribeLogGroups(&cloudwatchlogs.DescribeLogGroupsInput{
+		LogGroupNamePrefix: aws.String(awslogsLogGroupName),
+	})
+	require.NoError(t, err, "CloudWatchLogs describe log groups failed")
+	logGroupExists := false
+	for i := 0; i < len(respDescribeLogGroups.LogGroups); i++ {
+		if *respDescribeLogGroups.LogGroups[i].LogGroupName == awslogsLogGroupName {
+			logGroupExists = true
+			break
+		}
+	}
+
+	if !logGroupExists {
+		_, err := cwlClient.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
+			LogGroupName: aws.String(awslogsLogGroupName),
+		})
+		require.NoError(t, err, fmt.Sprintf("Failed to create log group %s", awslogsLogGroupName))
+	}
+
+	//parameterName := "FunctionalTest-SecretSupportLogDriverAwslogsMultiline"
+	parameterName := "FunctionalTestSecretSupportLogDriverAwslogsMultiline"
+	secretName := "awslogs-multiline-pattern"
+	ssmClient := ssm.New(session.New(), aws.NewConfig().WithRegion(*ECS.Config.Region))
+	input := &ssm.PutParameterInput{
+		Description: aws.String("Resource created for the ECS Agent Functional Test: TestLogDriverSecretSupport"),
+		Name:        aws.String(parameterName),
+		Value:       aws.String("^INFO"),
+		Type:        aws.String("String"),
+	}
+
+	// create parameter in parameter store if it does not exist
+	_, err = ssmClient.PutParameter(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ssm.ErrCodeParameterAlreadyExists:
+				t.Logf("Parameter %v already exists in SSM Parameter Store", parameterName)
+				break
+			default:
+				require.NoError(t, err, "SSM PutParameter call failed")
```

Comment:
can you use t.Errorf instead to make the logic clear that it will error if execute to this line. since you already check that err != nil

Reply:
Since this block of code resides in several test cases being involved in creating parameter in parameter store if not exist, I will leave this for now and try to optimize it all together later on.